### PR TITLE
[Bonmin] rebuild using latest versions

### PR DESCRIPTION
**Pull request by odow**
@tkralphs that seems to have done the trick:

```Julia
julia> using JuMP, AmplNLWriter, Bonmin_jll

julia> model = Model(() -> AmplNLWriter.Optimizer(Bonmin_jll.amplexe))
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: AmplNLWriter

julia> @variable(model, x >= 0)
x

julia> @objective(model, Min, x)
x

julia> optimize!(model)
Bonmin 1.8.8 using Cbc 2.10.5 and Ipopt 3.13.4
bonmin: 
Cbc3007W No integer variables - nothing to do

******************************************************************************
This program contains Ipopt, a library for large-scale nonlinear optimization.
 Ipopt is released as open source code under the Eclipse Public License (EPL).
         For more information visit https://github.com/coin-or/Ipopt
******************************************************************************

NLP0012I 
              Num      Status      Obj             It       time                 Location
NLP0014I             1         OPT 0        2 0.004643
Cbc3007W No integer variables - nothing to do

 	"Finished"

julia> value(x)
0.0
```

### CODE REVIEW

**Comment by tkralphs** [comment]
Cool, glad we could clear that up! I sure would like how it is possible to create such a weird mutant binary that seemed to be working and didn't crash, but also gave completely incorrect results with no real indication of a problem. It really is hard to imagine how that could happen...

**Comment by odow** [comment]
I would also like to know! But getting this working has already taken me longer than expected, and I'm not in a hurry to go debugging again...

File: C/Coin-OR/Bonmin/build_tarballs.jl [hunk 1]
```diff
@@ -50,10 +50,6 @@ make install
 rm "${includedir}/coin-or/IpoptConfig.h"
 """
 
-
-platforms = supported_platforms()
-filter!(!Sys.isfreebsd, platforms)
-platforms = expand_cxxstring_abis(platforms)
 filter!(x -> cxxstring_abi(x) != "cxx03", platforms)
 platforms = expand_gfortran_versions(platforms)
```
Comment (giordano) [inline C/Coin-OR/Bonmin/build_tarballs.jl:54]: I don't think this is necessary, as far as I can see from the audit log
```suggestion
```

Comment (odow) [reply C/Coin-OR/Bonmin/build_tarballs.jl]: Ah. We see `locks us to cxx11` but not `locks us to gfortran5`? I guess the auditor could check if the fortran versions were expanded, but it didn't see a lock and throw a warning?

Comment (giordano) [reply C/Coin-OR/Bonmin/build_tarballs.jl]: > I guess the auditor could check if the fortran versions were expanded, but it didn't see a lock and throw a warning?

That's something I'd like to be able to do, but at the moment it isn't possible: we'd need to collect all warnings (which we can't do) and check whether we built for a specific libgofrtran version and there were no warnings about it (similarly for C++ string ABI).  We need to rework the whole auditing process to be able to catch this.

**Pull request merged**